### PR TITLE
Add feature to assign and unassign tags to contacts

### PR DESCRIPTION
**Pull request by glennljw**
Fixes 3/6, 4/6 and 5/6 of #39.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/UnassignTagCommand.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+     * Removes a specific {@code Tag} from the {@code Set<Tag>} of a {@code Person}.
+     * @param personToEdit the person to remove the tag from.
+     * @param newTag the tag to remove.
+     * @return the person with the removed tag.
+     */
+    private static Person addTagToNewPerson(Person personToEdit, Tag newTag) {
+        requireNonNull(personToEdit);
+        requireNonNull(newTag);
+        assert personToEdit != null;
+        assert newTag != null;
```
Comment (takufunkai) [inline src/main/java/seedu/address/logic/commands/UnassignTagCommand.java:90]: I think having both `requireNonNull` and `assert` is redundant! If either one is null, requireNonNull would've thrown an error.

File: src/test/java/seedu/address/logic/commands/AssignTagCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getPersonList().size());
+
+        AssignTagCommand assignTagCommand = new AssignTagCommand(outOfBoundIndex, VALID_TAGNAME_FRIENDS);
+        String expectedMessage = Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+        assertCommandFailure(assignTagCommand, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_invalidIndexUncreatedTag_throwsCommandException() {
```
Comment (takufunkai) [inline src/test/java/seedu/address/logic/commands/AssignTagCommandTest.java:84]: Not quite sure what `UncreatedTag` here refers to 🤔

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 3]
```diff
@@ -140,6 +140,27 @@ public PersonBuilder withFavourite(String favourite) {
         return this;
     }
 
+    /**
+<<<<<<< HEAD
```
Comment (takufunkai) [inline src/test/java/seedu/address/testutil/PersonBuilder.java:144]: I think you forgot to remove this line!

**Review by takufunkai — CHANGES_REQUESTED** [review]
Overall a decent piece of work. Minor changes required, otherwise LGTM!

File: src/main/java/seedu/address/logic/commands/UnassignTagCommand.java (hunk 1 above)
Comment (glennljw) [reply src/main/java/seedu/address/logic/commands/UnassignTagCommand.java:90]: True, I'll fix that!

File: src/test/java/seedu/address/testutil/PersonBuilder.java (hunk 3 above)
Comment (glennljw) [reply src/test/java/seedu/address/testutil/PersonBuilder.java]: Thanks for catching this

**Review by takufunkai — APPROVED** [review]
LGTM!

**Pull request merged**